src/outputs/output_utils.hpp

```diff
@@ -41,31 +46,89 @@ namespace parthenon {
 namespace OutputUtils {
 // Helper struct containing some information about a variable
 struct VarInfo {
+ public:
+  static constexpr int VNDIM = MAX_VARIABLE_DIMENSION;
   std::string label;
   int num_components;
-  int nx6;
-  int nx5;
-  int nx4;
-  int nx3;
-  int nx2;
-  int nx1;
   int tensor_rank; // 0- to 3-D for cell-centered variables, 0- to 6-D for arbitrary shape
                    // variables
   MetadataFlag where;
   bool is_sparse;
   bool is_vector;
+  IndexShape cellbounds;
   std::vector<std::string> component_labels;
-  int Size() const { return nx6 * nx5 * nx4 * nx3 * nx2 * nx1; }
+  std::vector<TopologicalElement> topological_elements;
+  int ntop_elems;
+  bool element_matters;
```

Comment:
It would be useful to comment on what this denotes. I think it is used to store info about the topological element size?

Reply:
:+1: